packages/merge/src/typedefs-mergers/merge-typedefs.ts

```diff
@@ -185,26 +234,29 @@ function visitTypeSources(
         allDirectives,
         allNodes,
         visitedTypeSources,
+        repeatableLinkImports,
       );
     } else {
       throw new Error(
         `typeDefs must contain only strings, documents, schemas, or functions, got ${typeof typeSource}`,
       );
     }
   }
-  return { allDirectives, allNodes };
+  return { allDirectives, allNodes, repeatableLinkImports };
 }
 
 export function mergeGraphQLTypes(typeSource: TypeSource, config: Config): DefinitionNode[] {
   resetComments();
 
-  const { allDirectives, allNodes } = visitTypeSources(typeSource, config);
+  const { allDirectives, allNodes, repeatableLinkImports } = visitTypeSources(typeSource, config);
 
   const mergedDirectives = mergeGraphQLNodes(allDirectives, config) as Record<
     string,
     DirectiveDefinitionNode
   >;
 
+  config.repeatableLinkImports = repeatableLinkImports;
```

Comment:
This may not be how we want to pass this value around. I considered passing another argument to all the merge functions, but adding a config option was much simpler. Let me know if this would be better to do in a different way.